src/main/java/org/spongepowered/common/datapack/DataPackSerializer.java

```diff
@@ -44,12 +44,17 @@ public DataPackSerializer(final String token, final String typeDirectoryName) {
         this.typeDirectoryName = typeDirectoryName;
     }
 
-    protected void serialize(final Path dataPackDirectory, final List<T> objects) throws IOException {
-        FileUtils.deleteDirectory(dataPackDirectory.toFile());
+    protected boolean serialize(final Path dataPackDirectory, final List<T> objects) throws IOException {
+        final Path datapackDir = dataPackDirectory.resolve(this.getPackName());
```

Comment:
Should plugin generated content really be outside others? What is the reasoning for this? Datapack content is namespaced...

Reply:
wdym?
atm. this creates 2 datapacks:
`plugin-advancements` and `plugin-recipes` which contains the pack.mcmeta
and under `data/<namespace>/<type>/<value>.json` the recipe/advancement files